runtime/android/core/src/org/xwalk/core/XWalkActivityDelegate.java

```diff
@@ -10,10 +10,40 @@
 
 import org.xwalk.core.XWalkLibraryLoader.ActivateListener;
 import org.xwalk.core.XWalkLibraryLoader.DecompressListener;
-import org.xwalk.core.XWalkUpdater.XWalkUpdateListener;
 
 public class XWalkActivityDelegate
-            implements DecompressListener, ActivateListener, XWalkUpdateListener {
+            implements DecompressListener, ActivateListener {
+    /**
+     * Interface used to listen to XWalk background update events
+     */
+    public interface XWalkUpdateListener {
```

Comment:
I can't see the necessity of this interface, most of it aren't being used by other classes, and it could confuse other developers by the same name with `XWalkUpdater`